### ISSUE TITLE
No changelists listed in source control pane

### ISSUE THREAD

**Issue by allender** [mjcrouch/vscode-perforce#50]
**Describe the bug:**
I am unable to see any changelists in the source control panel.  I have verified that with the original version (off which this version is forked) works properly.  Here is a screenshot from the source control panel:

![2020-03-02 10_54_31-Settings](https://user-images.githubusercontent.com/1638265/75698860-131fc680-5c75-11ea-8a2a-c00b3d453b0a.png)

![perforcelog](https://user-images.githubusercontent.com/1638265/75699122-87f30080-5c75-11ea-880b-73d454700379.png)

I checked the log output and it looks like the describe and fstat commands are getting called correctly.  I see them in the log, but again, no changelist information shows directly in the source control panel.

**Expected behavior:**
We should be able to see all of the changelists and all of the files

**To Reproduce:**

**Versions & Details:**
 - Perforce extension:  3.6
 - VSCode version: 1.42.1
 - Operating System: Windows 10
 - Perforce server details (if known):
 - Are you using a multi-root workspace?: Np

**Additional context**

<details>

```
Please open view -> Output, select "Perforce Log" from the dropdown and paste it here, if possible / relevant
```

</details>

**Comment by mjcrouch** [comment]
hmm, just to make absolutely sure - you disabled the slevesque version? (most likely you have as we can't see any output from it!)

In the command palette can you run "Toggle developer tools", which will open the chrome dev console. Can you check if there are any exceptions logged there?

Did you use any versions of the fork prior to 3.6.x? (which I pushed on saturday IIRC)

**Comment by allender** [comment]
I've done all of the disabling/enabling multiple times to make sure.  I did check the console.  There are indeed errors.  Pasted here!
`console.ts:137 [Extension Host] rejected promise not handled within 1 second: TypeError: Cannot read property 'replace' of undefined
t.log @ console.ts:137
console.ts:137 [Extension Host] stack trace: TypeError: Cannot read property 'replace' of undefined	at Function.file (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:128:23)	at t.Model.makeResourceForShelvedFile (c:\Users\Mark.Allender\.vscode\extensions\mjcrouch.perforce-3.6.1\dist\extension.js:7:25402)	at c:\Users\Mark.Allender\.vscode\extensions\mjcrouch.perforce-3.6.1\dist\extension.js:7:25768	at Array.map (<anonymous>)	at c:\Users\Mark.Allender\.vscode\extensions\mjcrouch.perforce-3.6.1\dist\extension.js:7:25756	at Array.flatMap (<anonymous>)	at t.Model.<anonymous> (c:\Users\Mark.Allender\.vscode\extensions\mjcrouch.perforce-3.6.1\dist\extension.js:7:25724)	at Generator.next (<anonymous>)	at s (c:\Users\Mark.Allender\.vscode\extensions\mjcrouch.perforce-3.6.1\dist\extension.js:7:8436)	at processTicksAndRejections (internal/process/task_queues.js:89:5)
t.log @ console.ts:137
mainThreadExtensionService.ts:65 [[object Object]]Cannot read property 'replace' of undefined
$onExtensionRuntimeError @ mainThreadExtensionService.ts:65
mainThreadExtensionService.ts:66 TypeError: Cannot read property 'replace' of undefined
	at Function.file (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:128:23)
	at t.Model.makeResourceForShelvedFile (c:\Users\Mark.Allender\.vscode\extensions\mjcrouch.perforce-3.6.1\dist\extension.js:7:25402)
	at c:\Users\Mark.Allender\.vscode\extensions\mjcrouch.perforce-3.6.1\dist\extension.js:7:25768
	at Array.map (<anonymous>)
	at c:\Users\Mark.Allender\.vscode\extensions\mjcrouch.perforce-3.6.1\dist\extension.js:7:25756
	at Array.flatMap (<anonymous>)
	at t.Model.<anonymous> (c:\Users\Mark.Allender\.vscode\extensions\mjcrouch.perforce-3.6.1\dist\extension.js:7:25724)
	at Generator.next (<anonymous>)
	at s (c:\Users\Mark.Allender\.vscode\extensions\mjcrouch.perforce-3.6.1\dist\extension.js:7:8436)
	at processTicksAndRejections (internal/process/task_queues.js:89:5)`

I think that is all you need (tried to strip out some things from the log).  Let me know if you need more info.  Can take it offline as well if you'd like

**Comment by mjcrouch** [comment]
Thanks, I can see the line that's causing the issue - Do you have shelved files open for something other than edit?

It's doing the fstat to try get the details for a shelved file, and then (probably incorrectly) assuming that every file has a "clientFile" output in the shelved list, but clearly it doesn't - so I think there's a case I've missed for shelved files - unless there's osmething else wrong with the parsing

**Comment by mjcrouch** [comment]
Have reproduced this by shelving a file and then changing my p4 client so it doesn't include the client file, so quite likely that's the issue - will push a new version shortly

**Comment by mjcrouch** [comment]
Published 3.6.2 - Thanks for the help - let me know if it still has trouble!

**Comment by allender** [comment]
Huzzah!  works great.  thanks.
